src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -37,10 +37,31 @@ public static bool RequiresReflectionMethodBodyScannerForCallSite (LinkContext c
 			if (methodDefinition == null)
 				return false;
 
-			return
-				GetIntrinsicIdForMethod (methodDefinition) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel ||
+			bool isPInvokeImpl = methodDefinition.IsPInvokeImpl;
+			if (isPInvokeImpl) {
+				// HACK - we don't want to scan C++/CLI assemblies
+				if (IsCPPCLIAssembly (methodDefinition.DeclaringType.Module)) {
+					isPInvokeImpl = false;
+				}
+				// @TODO - there must be a better way to find out methods from C++/CLI assemblies
+				// Revisit after discussing with the team
```

Comment:
The proper way would be to detect the things that are actually problematic for us:

* the assembly has a p/invoke with a null entrypoint (indicates we're calling native code within the current module), or
* there's any `vtfixup`s (see ECMA-335 spec).

The p/invoke check is not very cheap in Cecil (not sure if we can enumerate all the p/invokes in the module). It is cheap to check with System.Reflection.Metadata.

This is probably the best we can do right now.

Neither of those would detect things like https://twitter.com/MStrehovsky/status/1384221192702218250 which IL Linker is also going to damage, but we probably have to just live with that.